File: CS50/sentimental-mario-more/mario.py
from cs50 import get_int


def makePyramid(h):
    if h == 0:
        return
    makePyramid(h-1)
    i = h
    while (i != height):
        print(' ', end='')
        i += 1
    for i in range(h):
        print('#', end='')
    print('  ', end='')
    for i in range(h):
        print('#', end='')
    print('')


while True:
    height = get_int('Height: ')
    if height > 0 and height < 9:
        break
makePyramid(height)

File: CS50/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

void makePyramid(int n, int x);

int main(void)
{
    int n;
    do
    {
        n = get_int("Height: ");
    }
    while (8 < n || n < 1);

    int x = n;

    makePyramid(n, x);

    return 0;
}

void makePyramid(int n, int x)
{
    if (n == 0)
    {
        return;
    }


    makePyramid(n - 1, x);

    for (int t = x; t > n; t--)
    {
        printf(" ");
    }

    for (int t = 0; t < n; t++)
    {
        printf("#");
    }

    printf("  ");

    for (int t = 0; t < n; t++)
    {
        printf("#");
    }

    printf("\n");

}